core/opendaq/opendaq/include/opendaq/instance_builder.h

```diff
@@ -0,0 +1,249 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/stringobject.h>
+#include <opendaq/instance.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_devices
+ * @addtogroup opendaq_instance InstanceBuilder
+ * @{
+ */
+
+/*!
+ * @brief Builder component of Instance objects. Contains setter methods to configure the Instance parameters, such as Context (Logger, Scheduler, ModuleManager) and RootDevice.
+ * Contains a  `build` method that builds the Instance object.
+ *
+ * The InstanceBuilder provides a fluent interface for setting various configuration options for an Instance
+ * object before its creation. It allows customization of the logger, module manager, scheduler and root device. 
+ * Once configured, calling the `build` method returns a fully initialized Instance object with the specified settings.
+ * 
+ * @subsection Configuration Methods:
+ * The InstanceBuilder provides the following configuration methods:
+ *
+ * - **Logger:** The custom Logger for the Instance. This logger will be used for logging messages related to
+ *   the Instance and its components. When configured, the GlobalLogLevel, ComponentLogLevel and LoggerSink will be ignored, as they are in use only with the default Instance logger
+ *
+ * - **GlobalLogLevel:** The default Logger global log level for the Instance. All log messages with a severity
+ *   level equal to or higher than the specified level will be processed.
+ *
+ * - **ComponentLogLevel:** The Logger level for a specific component of the Instance. Log messages related to
+ *   that component will be processed according to the specified log level.
+ *
+ * - **LoggerSink:** The logger sink to the default Instance logger. This sink will be responsible for processing
+ *   log messages, such as writing them to a file or sending them to a remote server.
+ *
+ * - **ModuleManager:** The custom ModuleManager for the Instance. When configured, the default module manager path
+ *   will be ignored.
+ *
+ * - **ModulePath:** The path for the default ModuleManager of the Instance. If a custom module manager has not
+ *   been set, this path will be used to load modules.
+ *
+ * - **Scheduler:** The custom scheduler for the Instance. If set, the number of worker threads will be ignored.
+ *
+ * - **SchedulerWorkerNum:** The number of worker threads in the scheduler of the Instance. If a scheduler has
+ *   not already been set, this defines the number of threads available for parallel processing.
+ *
+ * - **DefaultRootDeviceName:** The virtual Client as the default root device with the specified name. If the
+ *   RootDevice has not been set, the Instance Root Device will also be set as the virtual Client.
```

Comment:
The wording here is a bit unclear. What is the "virtual Client"? This should clearly state that the name of the default client root device is modified when this is set.

Also, should this not be "Default root device local ID"?